plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/components/upgrade-notice/index.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Notice, Button } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { recordEvent } from '@woocommerce/tracks';
+import { dispatch, select } from '@wordpress/data';
+import { findBlock } from '@woocommerce/utils';
+import {
+	createBlock,
+	// @ts-expect-error Type definitions for this function are missing in Gutenberg
+	createBlocksFromInnerBlocksTemplate,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import metadata from '../../block.json';
+import getInnerBlocksTemplate from '../../../../add-to-cart-with-options/utils/get-inner-blocks-template';
+
+const upgradeToBlockifiedAddtoCartWithOptions = async (
+	blockClientId: string
+) => {
+	const blocks = select( 'core/block-editor' ).getBlocks();
+	const foundBlock = findBlock( {
+		blocks,
+		findCondition: ( block ) =>
+			block.name === metadata.name && block.clientId === blockClientId,
+	} );
+
+	if ( ! foundBlock ) {
+		return;
+	}
+
+	const newBlock = createBlock(
+		'woocommerce/add-to-cart-with-options',
+		{
+			isDescendentOfSingleProductBlock:
+				foundBlock.attributes.isDescendentOfSingleProductBlock,
+		},
+		createBlocksFromInnerBlocksTemplate(
+			getInnerBlocksTemplate(
+				foundBlock.attributes.quantitySelectorStyle
+			)
+		)
+	);
+	dispatch( 'core/block-editor' ).replaceBlock(
+		foundBlock.clientId,
+		newBlock
+	);
+};
+
+export const UpgradeNotice = ( {
+	blockClientId,
+}: {
+	blockClientId: string;
+} ) => {
+	const notice = createInterpolateElement(
+		__(
+			'Upgrade the Add to Cart with Options block to <strongText /> for more features!',
+			'woocommerce'
+		),
+		{
+			strongText: (
+				<strong>
+					{ __( `a new blockified experience`, 'woocommerce' ) }
+				</strong>
+			),
+		}
+	);
+
+	const buttonLabel = __(
+		'Upgrade to the blockified Add to Cart with Options block',
+		'woocommerce'
+	);
+
+	const handleClick = () => {
+		upgradeToBlockifiedAddtoCartWithOptions( blockClientId );
+		recordEvent( 'blocks_add_to_cart_with_options_migration', {
+			transform_to: 'blockified',
+		} );
+	};
+
+	return (
+		<Notice isDismissible={ false }>
+			<>{ notice }</>
+			<br />
+			<br />
+			<Button variant="link" onClick={ handleClick }>
+				{ buttonLabel }
+			</Button>
+		</Notice>
+	);
+};
```

Comment:
Can't help but notice the similarity between:
- plugins/woocommerce-blocks/assets/js/blocks/product-elements/add-to-cart-form/components/upgrade-notice/index.tsx
- plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/components/downgrade-notice/index.tsx
- https://github.com/woocommerce/woocommerce/blob/c7648abe5b5e1387c9e8ead887f1913b75e5443a/plugins/woocommerce-blocks/assets/js/blocks/product-query/inspector-controls/upgrade-notice.tsx#L9
- https://github.com/woocommerce/woocommerce/blob/c7648abe5b5e1387c9e8ead887f1913b75e5443a/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/summary/upgrade.tsx#L38
- https://github.com/woocommerce/woocommerce/blob/c7648abe5b5e1387c9e8ead887f1913b75e5443a/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/upgrade-notice.tsx#L33

For some of them I am of course guilty of repeating the same code 😅 Not the scope of this PR but we should consider extracting the notice as a separate block. Not expecting it in this PR but just mentioning for transparency.

Reply:
Oh, good catch, I wasn't aware we had that many! Fully agree that we should unify them. :100:  I opened an issue here: https://github.com/woocommerce/woocommerce/issues/54076.